x/crosschain/keeper/keeper_cross_chain_tx_vote_inbound_tx.go

```diff
@@ -19,13 +19,43 @@ import (
 // reached, the ballot is finalized. When a ballot is finalized, a new CCTX is
 // created.
 //
-// If the receiver chain is a ZetaChain, the EVM deposit is handled and the
-// status of CCTX is changed to "outbound mined". If EVM deposit handling fails,
-// the status of CCTX is chagned to 'aborted'.
+// If the receiver chain is ZetaChain, `HandleEVMDeposit` is called. If the
+// tokens being deposited are ZETA, `MintZetaToEVMAccount` is called and the
+// tokens are minted to the receiver account on ZetaChain. If the tokens being
+// deposited are gas tokens or ERC20 of a connected chain, ZRC20's `deposit`
+// method is called and the tokens are deposited to the receiver account on
+// ZetaChain. If the message is not empty, system contract's `depositAndCall`
+// method is also called and an omnichain contract on ZetaChain is executed.
+// Omnichain contract address and arguments are passed as part of the message.
+// If everything is successful, the CCTX status is changed to `OutboundMined`.
 //
-// If the receiver chain is a connected chain, the inbound CCTX is finalized
-// (prices and nonce are updated) and status changes to "pending outbound". If
-// the finalization fails, the status of CCTX is changed to 'aborted'.
+// If the receiver chain is a connected chain, the `FinalizeInbound` method is
+// called to prepare the CCTX to be processed as an outbound transaction. To
+// cover the outbound transaction fee, the required amount of tokens submitted
+// with the CCTX are swapped using a Uniswap pool on ZetaChain for the ZRC20 of
+// the gas token of the receiver chain. The ZRC20 tokens are then burned. The
+// nonce is updated. If everything is successful, the CCTX status is changed to
+// `PendingOutbound`.
+//
+// ```mermaid
```

Comment:
Why not putting a link to the MD file instead of copying it into the comment? The Mermaid diagram is not really readable as text

Same comment for outbound file

Reply:
I agree that it takes too much space, but the idea is that it should be closer to code, so when something changes it can be immediatly reflected in the docs.

Reply:
I may have not gotten what you mean. Are the comments synchronized with the MD file?

Reply:
Yes, `make specs` [looks up](https://github.com/zeta-chain/node/blob/0f022af0732ec5d60461b09516fdc196c670c603/scripts/gen-spec.go) all messages in proto files, finds their implementation in Go code, then uses comments as message descriptions and generates markdown files. So, yes, keeper method comments are used to generate `messages.md`.

Reply:
We can have a separate (non-generated) `status.md` with the diagrams and remove diagrams from the code, but then when something in code changes, we might forget to update the diagrams. Maybe I'm overthinking this.

Reply:
OK